File: patterns/bewegungssensor-timer/rules-python/bewegungssensor_timer_gruppe.py
import threading

from openhab import rule, Registry
from openhab.triggers import GroupCommandTrigger

TIMEOUT_SECONDS = 5 * 60
timers = {}


@rule(triggers=[GroupCommandTrigger("gMotionDetectors", "ON")])
class AMotionDetectorTriggered:
    def execute(self, module, input):
        event = input.get("event")
        if not event:
            return

        item_name = event.getItemName()

        if item_name in timers:
            timers[item_name].cancel()

        def turn_off(name=item_name):
            Registry.getItem(name).sendCommand("OFF")
            del timers[name]

        t = threading.Timer(TIMEOUT_SECONDS, turn_off)
        t.start()
        timers[item_name] = t
        self.logger.info(item_name + ": Timer gestartet (" + str(TIMEOUT_SECONDS) + "s)")

File: patterns/bewegungssensor-timer/rules-js/bewegungssensor-timer-gruppe.js
const timers = new Map();
const TIMEOUT_MINUTES = 5;

rules.JSRule({
  name: "A Motion Detector triggered",
  triggers: [triggers.GroupCommandTrigger("gMotionDetectors", "ON")],
  execute: (event) => {
    const triggeringItem = items.getItem(event.itemName);
    const existing = timers.get(triggeringItem.name);

    if (!existing || existing.hasTerminated()) {
      console.log(triggeringItem.name + ": Timer gestartet (" + TIMEOUT_MINUTES * 60 + "s)");
      const t = actions.ScriptExecution.createTimer(
        time.ZonedDateTime.now().plusMinutes(TIMEOUT_MINUTES),
        () => {
          triggeringItem.sendCommand("OFF");
          timers.delete(triggeringItem.name);
        }
      );
      timers.set(triggeringItem.name, t);
    } else {
      existing.reschedule(time.ZonedDateTime.now().plusMinutes(TIMEOUT_MINUTES));
    }
  }
});
